### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by ndikumanaisaie**
### Config & basic setup: 

- Create React App.
- Installed [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
- Downloaded the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Created routes and view components (Rockets, Missions, and My Profile]). Used `<NavLink />` for the page navigation links and styled active class to indicate which section/page the user is currently on (underline active navigation link).
- Created directories for all application logic.

### Redux: Fetch data and update the Redux store

- Upon the first render fetched data from the SpaceX API endpoints.
- Once the data are fetched, dispatched an action to store the selected data in the Redux store.
- Actions are dispatched once and data are not added to the store on every re-render (i.e. when changing views / using navigation).
- Rockets is the default view, the rockets data is fetched when the application starts.
- The mission data is fetched (once) when a user navigates to the Missions section.

### Render UI: lists

- Used useSelector(), Redux Hook, to select the state slices and rendered lists of rockets and missions in corresponding routes.

### Render UI: conditional components rendering

- Rockets that have already been reserved show a "Reserved" badge and a "Cancel reservation" button instead of the default "Reserve rocket" (as per design).
- Missions that the user has joined already show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
- Rockets and Missions use the React conditional rendering syntax.

### Render UI: My Profile section

- Composed two layouts and list ONLY the rockets reserved and missions joined by the user (as per design).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,67 @@
-# Getting Started with Create React App
+# space-travelers-hub
+This is  a web application for a company that provides commercial and scientific space travel services. The application allows users to book rockets and join selected space missions.
 
-This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+## Below is the screenshot of my work
+![space-travelers-hub](https://github.com/ndikumanaisaie/space-travelers-hub/blob/use-Redux-in-React/src/assets/images/shot.png)
```
Comment (ghost) [inline README.md]: - [x] The link to your screenshot is broken, Kindly fix that to have a professional readme

File: README.md [hunk 2]
```diff
@@ -1,70 +1,67 @@
...
+- LinkedIn: [LinkedIn](https://linkedin.com/in/larabiu033)
 
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+## Link to URL on gitHub
+- [space-travelers-hub](https://github.com/arabiu033/space-travelers-hub.git)
 
-### `npm run build`
+## Link to URL in the browser
+- [space-travelers-hub](https://bookcollectionreact.netlify.app/)
```
Comment (ghost) [inline README.md]: - [x] Kindly provide the actual live demo of the project. Currently, you are using a different project link.

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi Team, 

Wonderful job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good Design
✅ All tests are passing👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (arabiu033) [reply README.md]: Done.

**Review by AmaduKamara — APPROVED** [review]
# Status: APPROVED 🟢🟢🟢 

Hi Team!

Good job on making the changes asked by the previous reviewer. 👍 

Your project is complete!  

ClappingLeonardoDicaprioGIF (4)

## Highlights:
✅ You used Gitflow correctly 👍 
✅ Linter checks passed  👍 
✅ Good commit messages  👍 
✅ Used Redux correctly 👍 
✅ App works fine as expected both locally and hosted link 👍 

There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**